compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -341,8 +342,25 @@ private void visitFunction(BLangFunction funcNode) {
             return;
         }
         boolean isNilableReturn = funcNode.symbol.type.getReturnType().isNullable();
+
+        BLangType returnTypeNode = funcNode.returnTypeNode;
+        if (returnTypeNode.type.tag == TypeTags.ARRAY && returnTypeNode instanceof BLangArrayType) {
```

Comment:
We can use node kind instead of the `instanceof` checks.

Eg - `funcNode.returnTypeNode.getKind()== NodeKind.ARRAY_TYPE`

Applicable to all other places.